### PR TITLE
Fix rp client registration logic.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issue #32.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,9 +30,10 @@ RUN openssl req \
...
     -nodes \
     -x509 \
-    -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com" \
+    -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=localhost" \
     -keyout privkey.pem \
     -out fullchain.pem
 
 EXPOSE 8443
-CMD ["node", "./bin/solid", "start"]
+ENV NODE_TLS_REJECT_UNAUTHORIZED=0
```
Comment (gobengo) [inline Dockerfile]: @dmitrizagidulin I don't think this is good to have as a default in the Dockerfile, as it opts-into allowing requests that might have a man-in-the-middle, and I don't think that's a good default for production deployments.

Even without this in here, it's still possible for someone to opt into the insecure option with `docker run -e NODE_TLS_REJECT_UNAUTHORIZED=0 interopalliance/life-server`

Comment (gobengo) [reply Dockerfile]: also this works:

```
docker run life-server bin/solid-test start
```

File: Dockerfile [hunk 2]
```diff
@@ -30,9 +30,10 @@ RUN openssl req \
...
+CMD ["node", "./bin/solid", "start", "--no-reject-unauthorized"]
```
Comment (gobengo) [inline Dockerfile]: same for this '--no-reject-unauthorized'. I think the defaults should be secure.

File: Dockerfile (hunk 1 above)
Comment (dmitrizagidulin) [reply Dockerfile]: You are absolutely right; I’ll change it.

**Pull request merged**